Gems/OpenXRVk/Code/Source/InputDeviceXRController.cpp

```diff
@@ -0,0 +1,424 @@
+    AZ::Vector2 InputDeviceXRController::Implementation::RawXRControllerState::GetRightThumbStickNormalizedValues() const
+    {
+        return AZ::Vector2(m_rightThumbStickXState / m_thumbStickMaxValue,
+                           m_rightThumbStickYState / m_thumbStickMaxValue);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceXRController::Implementation::ProcessRawControllerState([[maybe_unused]] const RawXRControllerState& rawControllerState)
+    {
+        // Update digital button channels...
+        for (const auto& [channelId, bitMask] : rawControllerState.m_buttonIdsToBitMasks)
+        {
+            const bool buttonState = (rawControllerState.m_digitalButtonStates & bitMask) != 0;
+            m_inputDevice.m_buttonChannelsById[channelId]->ProcessRawInputEvent(buttonState);
+        }
+
+        using xrc = InputDeviceXRController;
+
+        // Update the analog triggers...
+        const float triggerL = rawControllerState.GetLeftTriggerAdjustedForDeadZoneAndNormalized();
+        const float triggerR = rawControllerState.GetRightTriggerAdjustedForDeadZoneAndNormalized();
+        const float gripL = rawControllerState.GetLeftGripAdjustedForDeadZoneAndNormalized();
+        const float gripR = rawControllerState.GetRightGripAdjustedForDeadZoneAndNormalized();
+        m_inputDevice.m_triggerChannelsById[xrc::Trigger::LTrigger]->ProcessRawInputEvent(triggerL);
+        m_inputDevice.m_triggerChannelsById[xrc::Trigger::RTrigger]->ProcessRawInputEvent(triggerR);
+        m_inputDevice.m_triggerChannelsById[xrc::Trigger::LGrip]->ProcessRawInputEvent(gripL);
+        m_inputDevice.m_triggerChannelsById[xrc::Trigger::RGrip]->ProcessRawInputEvent(gripR);
+
+        // Update thumb-stick channels...
+        const AZ::Vector2 leftThumbStick = rawControllerState.GetLeftThumbStickAdjustedForDeadZoneAndNormalized();
+        const AZ::Vector2 leftThumbStickPreDeadZone = rawControllerState.GetLeftThumbStickNormalizedValues();
+        const float leftStickUp = AZ::GetClamp(leftThumbStick.GetY(), 0.f, 1.f);
+        const float leftStickDown = fabsf(AZ::GetClamp(leftThumbStick.GetY(), -1.f, 0.f));
+        const float leftStickLeft = fabsf(AZ::GetClamp(leftThumbStick.GetX(), -1.f, 0.f));
+        const float leftStickRight = AZ::GetClamp(leftThumbStick.GetX(), 0.f, 1.f);
+        const AZ::Vector2 rightThumbStick = rawControllerState.GetRightThumbStickAdjustedForDeadZoneAndNormalized();
+        const AZ::Vector2 rightThumbStickPreDeadZone = rawControllerState.GetRightThumbStickNormalizedValues();
+        const float rightStickUp = AZ::GetClamp(rightThumbStick.GetY(), 0.f, 1.f);
+        const float rightStickDown = fabsf(AZ::GetClamp(rightThumbStick.GetY(), -1.f, 0.f));
+        const float rightStickLeft = fabsf(AZ::GetClamp(rightThumbStick.GetX(), -1.f, 0.f));
+        const float rightStickRight = AZ::GetClamp(rightThumbStick.GetX(), 0.f, 1.f);
```

Comment:
Yeah, getting proper ranges for the raw device is still needed, but once it's "normalized", they should all fall within these ranges.  That's the point of this part, to "sanitize" the data so that the values here would appear similar to a gamepad controller.